src/pages/patientView/clinicalInformation/PatientViewPageStore.ts

```diff
@@ -259,6 +272,64 @@ export class PatientViewPageStore {
         invoke: () => this.cosmicDataInvoke()
     });
 
+    readonly mutSigData = remoteData({
+        invoke: async () => {
+            const mutSigdata = await internalClient.getSignificantlyMutatedGenesUsingGET({studyId: this.studyId});
+            const byEntrezGeneId: MutSigData = mutSigdata.reduce((map:MutSigData, next:MutSig) => {
+                map[next.entrezGeneId] = { qValue: next.qValue };
+                return map;
+            }, {});
+            return byEntrezGeneId;
+        }
+    });
+
+
+    readonly hotspotData = remoteData({
+        await: ()=> [
+            this.mutationData
+        ],
+        invoke: () => {

```

Comment:
An `async` function would be nicer here. It should be okay to deal with this in another PR though. I will take care of this later.